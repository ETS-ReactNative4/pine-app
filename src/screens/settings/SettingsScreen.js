import React, { Component } from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import headerStyles from '../../styles/headerStyles';
import DoneButton from '../../components/DoneButton';
import SettingsGroup from '../../components/SettingsGroup';
import SettingsLink from '../../components/SettingsLink';
import BaseSettingsScreen from './BaseSettingsScreen';

@connect((state) => ({
  settings: state.settings
}))
export default class SettingsScreen extends Component {
  static navigationOptions = ({ screenProps }) => ({
    title: 'Settings',
    headerStyle: headerStyles.header,
    headerTitleStyle: headerStyles.title,
    headerBackTitle: null,
    headerRight: (<DoneButton onPress={screenProps.dismiss} />),

    // HACK: Hack to disable the back navigation on the initial settings screen.
    headerLeft: (<Text />),
    gestureResponseDistance: { horizontal: -1, vertical: 135 }
  });

  _showGeneralSettings() {
    const navigation = this.props.navigation;
    navigation.navigate('GeneralSettings');
  }

  _showSecurityAndPrivacySettings() {
    const navigation = this.props.navigation;
    navigation.navigate('SecurityAndPrivacySettings');
  }

  _showBitcoinUnit() {
    const navigation = this.props.navigation;
    navigation.navigate('BitcoinUnit');
  }

  _showBitcoinFeeSettings() {
    const navigation = this.props.navigation;
    navigation.navigate('BitcoinFeeSettings');
  }

  render() {
    const settings = this.props.settings;

    return (
      <BaseSettingsScreen>
        <SettingsGroup>
          <SettingsLink name='General' onPress={this._showGeneralSettings.bind(this)} />
          <SettingsLink name='Security and Privacy' onPress={this._showSecurityAndPrivacySettings.bind(this)} isLastItem={true} />
        </SettingsGroup>

        <SettingsGroup>
          <SettingsLink name='Bitcoin Display Unit' value={settings.bitcoin.unit} onPress={this._showBitcoinUnit.bind(this)} />
          <SettingsLink name='Bitcoin Transaction Fees' value={settings.bitcoin.fee.level} onPress={this._showBitcoinFeeSettings.bind(this)} isLastItem={true} />
        </SettingsGroup>
      </BaseSettingsScreen>
    );
  }
}

SettingsScreen.propTypes = {
  settings: PropTypes.object,
  screenProps: PropTypes.object,
  dispatch: PropTypes.func,
  navigation: PropTypes.any
};
