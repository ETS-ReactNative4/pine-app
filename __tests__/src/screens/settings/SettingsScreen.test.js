import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SettingsScreen from '../../../../src/screens/settings/SettingsScreen';

jest.mock('../../../../src/containers/ErrorModalContainer', () => 'ErrorModalContainer');

const storeMock = {
  getState: jest.fn(() => ({
    settings: {
      user: {
        profile: {
          pineAddress: 'timothy@pine.cash'
        }
      },
      api: {
        baseUrl: 'f02f296b-05f9-4db1-a35a-37a3eb3b3f56'
      },
      bitcoin: {
        unit: 'BTC',
        fee: {
          level: 'Normal',
          satoshisPerByte: 100
        }
      }
    },
    keys: {
      items: {}
    }
  })),
  dispatch: jest.fn(),
  subscribe: jest.fn()
};

const navigationMock = {
  navigate: jest.fn(),
  state: {
    params: {}
  }
};

describe('SettingsScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsScreen store={storeMock} navigation={navigationMock} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
