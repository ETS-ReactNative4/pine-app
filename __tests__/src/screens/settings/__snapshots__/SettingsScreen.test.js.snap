// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#EFEFF3",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingTop": 35,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#C8C8CC",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#C8C8CC",
                "borderTopWidth": 0.5,
                "marginBottom": 35,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                      "fontWeight": "400",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                General
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#8F8E94",
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                        "position": "absolute",
                        "right": 15,
                        "textAlign": "right",
                        "width": 250,
                      },
                      Object {
                        "right": 35,
                      },
                    ],
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#C7C7CC",
                      "fontSize": 20,
                      "paddingTop": 2,
                      "position": "absolute",
                      "right": 15,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  Object {
                    "borderBottomWidth": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                      "fontWeight": "400",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                Security and Privacy
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#8F8E94",
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                        "position": "absolute",
                        "right": 15,
                        "textAlign": "right",
                        "width": 250,
                      },
                      Object {
                        "right": 35,
                      },
                    ],
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#C7C7CC",
                      "fontSize": 20,
                      "paddingTop": 2,
                      "position": "absolute",
                      "right": 15,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#C8C8CC",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#C8C8CC",
                "borderTopWidth": 0.5,
                "marginBottom": 35,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                      "fontWeight": "400",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                Bitcoin Display Unit
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#8F8E94",
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                        "position": "absolute",
                        "right": 15,
                        "textAlign": "right",
                        "width": 250,
                      },
                      Object {
                        "right": 35,
                      },
                    ],
                  ]
                }
              >
                BTC
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#C7C7CC",
                      "fontSize": 20,
                      "paddingTop": 2,
                      "position": "absolute",
                      "right": 15,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  Object {
                    "borderBottomWidth": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                      "fontWeight": "400",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                Bitcoin Transaction Fees
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#8F8E94",
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                        "position": "absolute",
                        "right": 15,
                        "textAlign": "right",
                        "width": 250,
                      },
                      Object {
                        "right": 35,
                      },
                    ],
                  ]
                }
              >
                Normal
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#C7C7CC",
                      "fontSize": 20,
                      "paddingTop": 2,
                      "position": "absolute",
                      "right": 15,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <ErrorModalContainer />
</View>
`;
